File: Intermediate/Binary_Search.java
import java.util.Scanner;

public class Binary_Search {
	static Scanner s1 = new Scanner(System.in);
	
	public static int[] takeInput() {
		int size = s1.nextInt();
		int arr[] = new int[size];
		for (int i = 0; i < size; i++) {
			arr[i] = s1.nextInt();
		}
		return arr;
	}
	public static int binarySearch(int[] arr, int num){
	      int start = 0 , end = arr.length -1 ;
	      while(start <= end)
	      {
	        int mid = (start + end)/2;
	        if(arr[mid] < num)
	        {
	          start = mid+1;
	        }
	        else if(arr[mid] > num)
	        {
	          end = mid-1;
	        }
	        else if(arr[mid]==num)
	          return mid;
	      }
	      return -1 ;// if the number is not found 
		}
	public static void main(String[] args) {
		int[] arr = takeInput();
		int num = s1.nextInt();
		System.out.print(binarySearch(arr, num));
	}
}

File: Intermediate/BinarySearch.cpp

#include<iostream>
using namespace std;
bool binarysearch(int arr[],int start,int end,int k)
{
    if(start>end){
        return false;
    }
     int mid=(start+end)/2;

      if(arr[mid]==k){
         return true;
      }
        if(arr[mid]>k){
          return binarysearch(arr,start,mid-1,k);
        }

          return binarysearch(arr,mid+1,end,k);
}
  int main()
  {
      int n,i;
      cout<<"Enter the size of array"<<endl;
      cin>>n;
      int arr[1000];
      cout<<"Enter the elements of array"<<endl;

      for(i=0;i<n;i++){
          cin>>arr[i];
      }

       int k;
       cout<<"Enter the no to be found"<<endl;
       cin>>k;
       int start=0;
       int end=n-1;
       if(binarysearch(arr,start,end,k)){
           printf("FOUND");
       }
       else{
           printf("NOT FOUND");
       }
  }
